WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Edit Price/ProductPriceSettingsViewModelTests.swift

```diff
@@ -512,6 +512,26 @@ final class ProductPriceSettingsViewModelTests: XCTestCase {
         waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
     }
 
+    func test_completeUpdating_new_sale_with_nil_sale_price() {
+        // Arrange
+        let dateOnSaleStart = date(from: "2019-09-02T21:30:00")
+        let dateOnSaleEnd = date(from: "2019-09-27T21:30:00")
+        let product = Product.fake().copy(dateOnSaleStart: dateOnSaleStart, dateOnSaleEnd: dateOnSaleEnd, salePrice: nil)
+        let model = EditableProductModel(product: product)
+        let viewModel = ProductPriceSettingsViewModel(product: model)
+
+        // Act
+        let expectation = self.expectation(description: "Wait for error")
+        viewModel.completeUpdating(onCompletion: { (_, _, _, _, _, _, _) in
+            XCTFail("Completion block should not be called")
+        }, onError: { error in
+            // Assert
+            XCTAssertEqual(error, .newSaleWithEmptySalePrice)
+            expectation.fulfill()
+        })
+        waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
```

Comment:
We have some `waitFor` functions defined in https://github.com/woocommerce/woocommerce-ios/blob/develop/TestKit/Sources/TestKit/XCTestCase+Wait.swift that can help this test to be read in a more synchronous way.

EG:
https://github.com/woocommerce/woocommerce-ios/blob/4fe952211777fad134084cbc19d4db229a37c8c9/Networking/NetworkingTests/Remote/ProductAttributesRemoteTests.swift#L54-L68

Reply:
That's great! Thanks for sharing 🙏 The test looks super clean now!